baram/ec2_manager.py

```diff
@@ -1,105 +1,274 @@
 import boto3
+import traceback
+
+from baram.log_manager import LogManager
 
 
 class EC2Manager(object):
     def __init__(self):
         self.cli = boto3.client('ec2')
+        self.logger = LogManager.get_logger()
 
-    def list_security_groups(self):
-        '''
+    def list_sgs(self):
+        """
         Describes the specified security groups or all of your security groups.
 
         :return: SecurityGroups
-        '''
+        """
         return self.cli.describe_security_groups()['SecurityGroups']
 
+    def list_unused_sg_ids(self, description_filter: str = '', unused_domain_ids: list = []):
+        """
+        Describe useless and deletable security group ids.
+        (including disused SageMaker domain related things)
+
+        :param description_filter:
+        :param unused_domain_ids:
+        :return: GroupIds
+        """
+        # 1. Get security groups not related to anything in ec2
+        sgs = self.list_sgs()
+        valid_sg_ids = set([pair['sg_id'] for pair in self.list_vpc_sg_eni_subnets()])
+        result = [sg['GroupId'] for sg in sgs
+                  if sg['GroupId'] not in valid_sg_ids
+                  or (description_filter in sg['Description']
+                      and sum([domain_id in sg['Description'] for domain_id in unused_domain_ids]) == 0)]
```

Comment:
Because `sg['Description']` looks like `[DO NOT DELETE] Security Group that allows outbound NFS traffic for SageMaker Notebooks Domain [d-qkeerztiyyjj]` and each `domain_id` looks like `d-qkeerztiyyjj`, this needs different remediation.

Reply:
I thought separating `domain_id` from the last part of `sg['Description']` at first time, but I didn't apply this because it's too heuristic. Is this would be better?

Reply:
let it be this time.